recipes/gcc/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+    def requirements(self):
+        self.requires("mpc/1.1.0")
+        self.requires('mpfr/4.0.2')
+        self.requires("gmp/6.1.2")
+        self.requires('zlib/1.2.11')
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows builds aren't supported (yet), sorry")
+        if tools.cross_building(self.settings):
+            raise ConanInvalidConfiguration("no cross-building support (yet), sorry")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "gcc-releases-gcc-%s" % self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    @property
+    def _make_args(self):
+        if self.settings.os == "Macos":
+            return ["BOOT_LDFLAGS=-Wl,-headerpad_max_install_names"]
+        return []
+
+    def build(self):
+        # If building on x86_64, change the default directory name for 64-bit libraries to "lib":
+        libdir = "%s/lib/gcc/%s" % (self.package_folder, self.version)
+        tools.replace_in_file(os.path.join(self.source_folder,
+                                           self._source_subfolder, "gcc", "config", "i386", "t-linux64"),
+                              "m64=../lib64", "m64=../lib", strict=False)
+        # Ensure correct install names when linking against libgcc_s;
+        # see discussion in https://github.com/Homebrew/legacy-homebrew/pull/34303
+        tools.replace_in_file(os.path.join(self.source_folder,
+                                           self._source_subfolder, "libgcc", "config", "t-slibgcc-darwin"),
+                              "@shlib_slibdir@", libdir, strict=False)
+        autotools = self._configure_autotools()
+        autotools.make(args=self._make_args)
+
+    def package_id(self):
+        del self.info.settings.compiler
+
+    def _remove_by_mask(self, pattern):
+        import fnmatch
+        for root, _, filenames in os.walk(self.package_folder):
+            for filename in filenames:
+                if fnmatch.fnmatch(filename, pattern):
+                    os.unlink(os.path.join(self.package_folder, root, filename))
+
+    def package(self):
+        autotools = self._configure_autotools()
+        autotools.install(args=self._make_args)
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        self._remove_by_mask("*.la")
+        self.copy(pattern="COPYING*", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        bindir = os.path.join(self.package_folder, "bin")
+        self.output.info("Appending PATH env var with : " + bindir)
+        self.env_info.PATH.append(bindir)
+
+        cc = os.path.join(bindir, "gcc-%s" % self.version)
+        self.output.info("Creating CC env var with : " + cc)
+        self.env_info.CC = cc
+
+        cxx = os.path.join(bindir, "g++-%s" % self.version)
+        self.output.info("Creating CXX env var with : " + cxx)
+        self.env_info.CXX = cxx
```

Comment:
Do we want to set `CC` and `CXX` here?
Isn't that the role of the profile?

Reply:
I assume that they are needed here, if you are using this recipe as a `build_requires` then you expect it to populate the environment so it is actually used...

Reply:
Yes in this case, as you require this specific package, you have to use this compiler. Using a CC and CXX from profile, pointing to another compiler, creates an impredecible behavior

Reply:
is there a way to set CC within the profile relative to the build requires?
something like:
```
[build_requires]
gcc/10.1.0@
[env]
CC=os.path.join(self.deps_cpp_info["gcc"].root_path, "bin", "gcc")
CXX=os.path.join(self.deps_cpp_info["gcc"].root_path, "bin", "g++")
```
I doubt it's possible right now, so I worry there is no way to achieve the same result without env_info

Reply:
Nope, but it's a relevant feature.

Reply:
@SSE4 
I think this is now possible with using conan 1.38.0, jusing python code in profiles.